### ISSUE TITLE
2.3.6 Beta1 Internal Error when adding an invalid rss feed

### ISSUE THREAD

**Issue by Cpuroast** [sabnzbd/sabnzbd#1194]
in 2.3.6Beta1, If you try to add an RSS feed with an invalid url you get this, probably an undesirable way of saying it's an invalid URL :)

500 Internal Server Error
The server encountered an unexpected condition which prevented it from fulfilling the request.

Traceback (most recent call last):
  File "cherrypy/_cprequest.pyc", line 670, in respond
  File "cherrypy/lib/encoding.pyc", line 220, in __call__
  File "cherrypy/_cpdispatch.pyc", line 60, in __call__
  File "sabnzbd/interface.pyc", line 127, in internal_wrap
  File "sabnzbd/interface.pyc", line 1760, in index
  File "sabnzbd/rss.pyc", line 65, in run_feed
  File "sabnzbd/decorators.pyc", line 36, in call_func
  File "sabnzbd/rss.pyc", line 304, in run_feed
  File "sabnzbd/utils/feedparser.pyc", line 375, in __getitem__
KeyError: 'href'

After the error, if you hit back and refresh, the feed does appear in the list, but if you click on it you get the same error as when you initially added it.

**Comment by thezoggy** [comment]
what is an example you were trying to add? what sensible thing can we do to prevent user error?

**Comment by Cpuroast** [comment]
I tried something like: ajgjagdjsgjags as in something completely random.

I don't think an internal server error is quite how we should handle this: something like a http://fqdn(or IP) or https://fqdn(or IP) validator and display like "Invalid URL" next to the field upon clicking add and obviously not actually adding the invalid url.

Something has obviously changed in the logic, as I was able to add random urls in the past to test things without the internal server error.
